src/main/webapp/index.html

```diff
@@ -100,8 +106,93 @@ <h1>Favorite SDKs</h1>
+            messagingSenderId: "412575630040",
+            appId: "1:412575630040:web:824b2d31f4e1f06b100dad",
+        };
+
+        // Initialize Firebase
+        firebase.initializeApp(firebaseConfig);
+
+        // Redirect to Google Sign In
+        function googleSignIn() {
+            firebase.auth().onAuthStateChanged(function(user) {
+            if (user) {
+                // User is signed in.
+            } else {
+                // No user is signed in.
+                var provider = new firebase.auth.GoogleAuthProvider();
+                firebase.auth().signInWithRedirect(provider);
+                firebase.auth().getRedirectResult().then(function(result) {
+                    if (result.credential) {
+                        // This gives you a Google Access Token. You can use it to access the Google API.
+                        var token = result.credential.accessToken;
+                    }
+                    // The signed-in user info.
+                    var user = result.user;
+                    }).catch(function(error) {
+                        var errorCode = error.code;
+                        var errorMessage = error.message;
+                        // The email of the user's account used.
+                        var email = error.email;
+                        // The firebase.auth.AuthCredential type that was used.
+                        var credential = error.credential;
+                    });
+                }
+            });
+        }
+           
+        // Sign user out of google sign-in
+        function googleSignOut() {
+            firebase.initializeApp(firebaseConfig);
+
+        // Redirect to Google Sign In
+            function googleSignIn() {
+              firebase.auth().onAuthStateChanged(function(user) {
+                if (user) {
+                    // User is signed in.
+                } else {
+                    // No user is signed in.
+                    var provider = new firebase.auth.GoogleAuthProvider();
+                    firebase.auth().signInWithRedirect(provider);
+                    firebase.auth().getRedirectResult().then(function(result) {
+                      if (result.credential) {
+                          // This gives you a Google Access Token. You can use it to access the Google API.
+                          var token = result.credential.accessToken;
+                      }
+                      // The signed-in user info.
+                      var user = result.user;
+                      }).catch(function(error) {
+                          var errorCode = error.code;
+                          var errorMessage = error.message;
+                          // The email of the user's account used.
+                          var email = error.email;
+                          // The firebase.auth.AuthCredential type that was used.
+                          var credential = error.credential;
+                          });
```

Comment:
Instead of repeating this code, is there any way you can move it to some method and call that here?